### PR TITLE
[FLINK-30604][runtime] Remove the limitation that parallelism decided by adaptive batch scheduler is always a power of two

### PR DESCRIPTION
**Pull request by wanglijie95**
## What is the purpose of the change

In [FLINK-26517](https://issues.apache.org/jira/browse/FLINK-26517), we add a limitation that the parallelism decided by adaptive batch scheduler must be 2^N(for the subpartitions can evenly distribute to downstream tasks). After [FLINK-29666](https://issues.apache.org/jira/browse/FLINK-29666), we can evenly distribute data to downstream tasks, so we can remove the limitation

## Verifying this change

This change is already covered by existing tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**no**)
  - The serializers: (**no**)
  - The runtime per-record code paths (performance sensitive): (**no**)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (**no**)
  - The S3 file system connector: (**no**)

## Documentation

  - Does this pull request introduce a new feature? (**no**)
  - If yes, how is the feature documented? ( **not applicable**)

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java [hunk 1]
```diff
@@ -582,14 +578,9 @@ public enum SchedulerType {
                     .withDescription(
                             Description.builder()
                                     .text(
-                                            "The average size of data volume to expect each task instance to process if %s has been set to %s. "
-                                                    + "Note that since the parallelism of the vertices is adjusted to a power of 2, "
-                                                    + "the actual average size will be 0.75~1.5 times this value. "
-                                                    + "It is also important to note that when data skew occurs or the decided parallelism reaches the %s (due to too much data), "
```
Comment (zhuzhurk) [inline flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java:588]: I think this note is still needed. 
Maybe with a bit improvement: "It is also important to note that" -> "Note that"

**Review by zhuzhurk — COMMENTED** [review]
Thanks for creating this PR.
It generally looks good to me except for one minor comment.

**Comment by wanglijie95** [comment]
@zhuzhurk Thanks for review. I 've addressed your comment, please take a look.

**Review by zhuzhurk — APPROVED** [review]
LGTM.

**Pull request closed without merging**